tests/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolverTest.php

```diff
@@ -0,0 +1,220 @@
+    }
+
+    public function testReturnsSetWithResolvedValue()
+    {
+        $value = new FixtureMethodCallValue(
+            $reference = new FakeValue(),
+            $functionCall = new FunctionCallValue('getFoo', [
+                $arg1 = new FakeValue(),
+            ])
+        );
+        $fixture = new FakeFixture();
+        $set = ResolvedFixtureSetFactory::create(new ParameterBag(['foo' => 'bar']));
+        $scope = ['val' => 'scopie'];
+        $context = new GenerationContext();
+        $context->markIsResolvingFixture('foo');
+
+        $valueResolverContext = new GenerationContext();
+        $valueResolverContext->markIsResolvingFixture('foo');
+        $valueResolverContext->markAsNeedsCompleteGeneration();
+
+        $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+        $dummyProphecy->getFoo('resolved_argument')
+            ->shouldBeCalled()
+            ->willReturn('resolved_value')
+        ;
+
+        /** @var DummyWithGetter $dummy */
+        $dummy = $dummyProphecy->reveal();
+
+        $valueResolverProphecy = $this->prophesize(ValueResolverInterface::class);
+        $valueResolverProphecy
+            ->resolve($arg1, $fixture, $set, $scope, $context)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet(
+                    'resolved_argument',
+                    $newSet = ResolvedFixtureSetFactory::create(new ParameterBag(['ping' => 'pong']))
+                )
+            )
+        ;
+        $valueResolverProphecy
+            ->resolve($reference, $fixture, $newSet, $scope, $valueResolverContext)
+            ->willReturn(
+                new ResolvedValueWithFixtureSet($dummy, $newSet)
+            )
+        ;
+        /** @var ValueResolverInterface $valueResolver */
+        $valueResolver = $valueResolverProphecy->reveal();
+
+        $expected = new ResolvedValueWithFixtureSet('resolved_value', $newSet);
+
+        $resolver = new FixtureMethodCallReferenceResolver($valueResolver);
+        $actual = $resolver->resolve($value, $fixture, $set, $scope, $context);
+
+        $this->assertEquals($expected, $actual);
+
+        $valueResolverProphecy->resolve(Argument::cetera())->shouldHaveBeenCalledTimes(2);
+        $dummyProphecy->getFoo(Argument::cetera())->shouldHaveBeenCalledTimes(1);
+    }
+
+    public function testThrowsAResolverExceptionOnError()
+    {
+        try {
+            $value = new FixtureMethodCallValue(
+                $reference = new MutableValue(new DummyWithGetter()),
+                $functionCall = new FunctionCallValue('getFoo')
+            );
+            $set = ResolvedFixtureSetFactory::create(new ParameterBag(['foo' => 'bar']));
+
+            $error = new \Error();
+            $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+            $dummyProphecy->getFoo()->will(function () use ($error) {
```

Comment:
Can't use `willThrow` here as Prophecy `ThrowPromise` cannot be constructed with Errors